Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp

```diff
@@ -91,31 +91,32 @@ static int GetTypeSize(MemoryViewWidget::Type type)
 
 static int GetCharacterCount(MemoryViewWidget::Type type)
 {
+  // Max number of characters +1 for spacing between columns.
   switch (type)
   {
   case MemoryViewWidget::Type::ASCII:
-    return 1;
-  case MemoryViewWidget::Type::Hex8:
     return 2;
-  case MemoryViewWidget::Type::Unsigned8:
+  case MemoryViewWidget::Type::Hex8:
     return 3;
+  case MemoryViewWidget::Type::Unsigned8:
+    return 4;
   case MemoryViewWidget::Type::Hex16:
   case MemoryViewWidget::Type::Signed8:
-    return 4;
-  case MemoryViewWidget::Type::Unsigned16:
     return 5;
-  case MemoryViewWidget::Type::Signed16:
+  case MemoryViewWidget::Type::Unsigned16:
     return 6;
+  case MemoryViewWidget::Type::Signed16:
+    return 7;
   case MemoryViewWidget::Type::Hex32:
-    return 8;
-  case MemoryViewWidget::Type::Float32:
     return 9;
+  case MemoryViewWidget::Type::Float32:
+    return 11;
   case MemoryViewWidget::Type::Double:
   case MemoryViewWidget::Type::Unsigned32:
   case MemoryViewWidget::Type::Signed32:
-    return 10;
+    return 12;
   default:
-    return 8;
+    return 10;
   }
```

Comment:
What's up with all the changes here?

Reply:
Can't `std::numeric_limits<T>::digits10` be used here?

Reply:
Interesting, I'll try that.

I was adding the extra space character into it, so I didn't have to calculate it multiple times. I'll fully rewrite it if numeric limits works.

Reply:
Documentation says numeric limits will be off-by-one for things like ints, because not all three digit numbers can be represented by an int8, for example.

I could have it give the max characters then go to a calculation to add spacing/padding, if that is clearer. There might be some other way to set the space between columns, but I don't know it.

Reply:
That's how `digits10` [works](https://stackoverflow.com/a/36631469) regarding it being off by one. Here is an example using SFINAE:
```c++
template<typename T,
         typename std::enable_if_t<std::is_integral_v<T>>* = nullptr>
constexpr int character_count()
{
  return 1 + std::numeric_limits<T>::digits10 + int(std::numeric_limits<T>::is_signed);
}

template<typename T,
         typename std::enable_if_t<std::is_floating_point_v<T>>* = nullptr>
constexpr int character_count();

template<>
constexpr int character_count<float>()
{
  // FLT_MAX = 3.40282e+38
  return 11;
}

template<>
constexpr int character_count<double>()
{
  // DBL_MAX = 1.79769e+308
  return 12;
}
```
I don't think we gain much using this approach, though.